cg/store/api/find_basic_data.py

```diff
@@ -106,10 +109,21 @@ def get_applications(self) -> List[Application]:
             .all()
         )
 
+    def get_application_versions(self) -> List[ApplicationVersion]:
+        """Return all application versions"""
+        return self._get_query(table=ApplicationVersion).all()
+
     def application_version(self, application: Application, version: int) -> ApplicationVersion:
-        """Fetch an application version."""
-        query = self.ApplicationVersion.query.filter_by(application=application, version=version)
-        return query.first()
+        """Return an application version."""
+        return apply_application_versions_filter(
+            application_versions=self._get_query(table=ApplicationVersion),
+            application=application,
+            version=version,
```

Comment:
It might be better to only send in the property we are filtering on instead of the entire object.

Reply:
Using only the IDs simplifies the filter functions and reduces the amount of data being passed around

Reply:
Included in another PR